zerver/tests/test_openapi.py

```diff
@@ -366,10 +369,27 @@ def check_argument_types(self, function: Callable[..., HttpResponse],
         OpenAPI data defines a different type than that actually accepted by the function.
         Otherwise, we print out the exact differences for convenient debugging and raise an
         AssertionError. """
-        openapi_params = set([(element["name"], VARMAP[element["schema"]["type"]]) for
-                             element in openapi_parameters])
+        openapi_params = set()  # type: Set[Tuple[str, Union[type, Tuple[type, type]]]]
+        for element in openapi_parameters:
+            name = element["name"]  # type: str
+            _type = VARMAP[element["schema"]["type"]]
+            if _type == list:
+                items = element["schema"]["items"]
+                if "anyOf" in items.keys():
+                    subtypes = []
+                    for st in items["anyOf"]:
+                        st = st["type"]
+                        subtypes.append(VARMAP[st])
+                    self.assertTrue(len(subtypes) > 1)
+                    sub_type = self.get_type_by_priority(subtypes)  # type: ignore # sub_type is not List[Optional[type]], it's a List[type], handled in above step.
```

Comment:
The reason it’s `List[Optional[type]]` is not that it could be empty, it’s that `VARMAP` contains `None`, which is not a `type`. Shouldn’t that have been `NoneType` (i.e. `type(None)`)?

Reply:
Makes sense. None is the value and NoneType is the type (just like what you said).